### DISCUSSION TITLE
Hung at initialising devices

### DISCUSSION THREAD

**Discussion by huckabeec** [ipxe/ipxe#1021] [Q&A]
![screenshot-ipxe](https://github.com/ipxe/ipxe/assets/9270816/98111512-a806-4542-b10e-08387fdd01d8)

Trying to boot a Dell R430 with snponly.efi built from git sources with:

`make DEBUG=init:3,rbg,drbg bin-x86_64-efi/snponly.efi`

It hangs at INIT startup devices and then reboots.

System is booting from an Intel 10GbE 2P X710 Adapter.  I have some R440s that have booted using the same card previously without issues, so not sure if why this is having issues.  

What else can I do to make this produce more debug info on what is going on ?

**Comment by NiKiZe** [comment]
What have you done to rule out other hardware issues such as bad RAM?

**Reply by huckabeec** [reply]
This system has been up for over two years - we’re pxe booting it now for reinstall purposes.
It has passed Dell’s diagnostic tests and has the latest firmware updates for all of the hardware.
No signs of any hardware issues. I’m happy to run some other diagnostic tests if you have suggestions.

**Reply by NiKiZe** [reply]
Oh a Dell, sorry missed that part, always buggy, are you sure you are running latest version of iPXE as well?
Sorry I don't have better debug info of hand but snp and nii, also efidrv (this might be incorrect as I'm on phone and verifying sources is a bit difficult)

**Reply by huckabeec** [reply]
I built iPXE straight from the github clone of the master branch - if there are newer sources I'm happy to try them.
I'll try the additional debug items, thanks,

**Comment by huckabeec** [comment]
<img width="619" alt="pic001" src="https://github.com/ipxe/ipxe/assets/9270816/dd40fa21-f14c-4da1-b160-cdf4cd281d75">
<img width="619" alt="pic002" src="https://github.com/ipxe/ipxe/assets/9270816/5ecc7545-e80a-4312-a3ae-2687e949e571">
<img width="619" alt="pic003" src="https://github.com/ipxe/ipxe/assets/9270816/d0ed97ec-fefd-468b-a2cd-0f463dd03945">
<img width="693" alt="pic004" src="https://github.com/ipxe/ipxe/assets/9270816/1213bd03-f734-417c-99a6-32dc1020add2">

Had to stitch these together from a recording of the boot over the iDRAC.  Am I missing something or is it not seeing the x710 card as supported ?  Could the recent firmware update for the card broken support ?

**Reply by huckabeec** [reply]
As an additional datapoint, I built ipxe.iso and booted from that - it had no issues and no freezing.
